glauth/CHANGELOG.md

```diff
@@ -0,0 +1,207 @@
+# Changelog for [unreleased] (UNRELEASED)
+
+The following sections list the changes in ocis-glauth unreleased.
+
+[unreleased]: https://github.com/owncloud/ocis/glauth/compare/v0.5.0...master
+
+## Summary
+
+* Bugfix - Return invalid credentials when user was not found: [#30](https://github.com/owncloud/ocis/glauth/pull/30)
+* Bugfix - Query numeric attribute values without quotes: [#28](https://github.com/owncloud/ocis/glauth/issues/28)
+* Bugfix - Use searchBaseDN if already a user/group name: [#214](https://github.com/owncloud/product/issues/214)
+* Bugfix - Fix LDAP substring startswith filters: [#31](https://github.com/owncloud/ocis/glauth/pull/31)
+
+## Details
+
+* Bugfix - Return invalid credentials when user was not found: [#30](https://github.com/owncloud/ocis/glauth/pull/30)
+
+   We were relying on an error code of the ListAccounts call when the username and password was
+   wrong. But the list will be empty if no user with the given login was found. So we also need to check
+   if the list of accounts is empty.
+
+   https://github.com/owncloud/ocis/glauth/pull/30
+
+
+* Bugfix - Query numeric attribute values without quotes: [#28](https://github.com/owncloud/ocis/glauth/issues/28)
+
+   Some LDAP properties like `uidnumber` and `gidnumber` are numeric. When an OS tries to look up a
+   user it will not only try to lookup the user by username, but also by the `uidnumber`:
+   `(&(objectclass=posixAccount)(uidnumber=20000))`. The accounts backend for glauth was
+   sending that as a string query `uid_number eq '20000'` in the ListAccounts query. This PR
+   changes that to `uid_number eq 20000`. The removed quotes allow the parser in ocis-accounts to
+   identify the numeric literal.
+
+   https://github.com/owncloud/ocis/glauth/issues/28
+   https://github.com/owncloud/ocis/glauth/pull/29
+   https://github.com/owncloud/ocis/accounts/pull/68
+
+
+* Bugfix - Use searchBaseDN if already a user/group name: [#214](https://github.com/owncloud/product/issues/214)
+
+   In case of the searchBaseDN already referencing a user or group, the search query was ignoring
+   the user/group name entirely, because the searchBaseDN is not part of the LDAP filters. We
+   fixed this by including an additional query part if the searchBaseDN contains a CN.
+
+   https://github.com/owncloud/product/issues/214
+   https://github.com/owncloud/ocis/glauth/pull/32
+
+
+* Bugfix - Fix LDAP substring startswith filters: [#31](https://github.com/owncloud/ocis/glauth/pull/31)
+
+   Filters like `(mail=mar*)` are currentld not parsed correctly, but they are used when
+   searching for recipients. This PR correctly converts them to odata filters like
+   `startswith(mail,'mar')`.
+
+   https://github.com/owncloud/ocis/glauth/pull/31
+
+# Changelog for [0.5.0] (2020-07-23)
+
+The following sections list the changes in ocis-glauth 0.5.0.
+
+[0.5.0]: https://github.com/owncloud/ocis/glauth/compare/v0.4.0...v0.5.0
```

Comment:
Codacy found an issue: [[final-definition] Move definitions to the end of the file (after the node at line `202`)](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6202733)